File: 3689. Maximum Total Subarray Value I.py
class Solution:
    def maxTotalValue(self, nums: List[int], k: int) -> int:

        # প্রথম element-কে initial minimum এবং maximum ধরা হচ্ছে
        minimum_ = maximum_ = nums[0]

        # পুরো array traverse করে minimum এবং maximum বের করছি
        for i in nums:
            minimum_ = min(minimum_, i)  # minimum update
            maximum_ = max(maximum_, i)  # maximum update

        # answer = (global max - global min) * k
        return (maximum_ - minimum_) * k

File: 3689. Maximum Total Subarray Value I.cpp
class Solution {
public:
    long long maxTotalValue(vector<int>& A, int k) {

        // প্রথম element-কে initial minimum এবং maximum ধরা হচ্ছে
        int gMin = A.front();
        int gMax = A.front();

        // পুরো array traverse করে global minimum এবং maximum বের করছি
        for (auto& n : A) {

            // minimum update
            gMin = min(gMin, n);

            // maximum update
            gMax = max(gMax, n);
        }

        // সর্বোচ্চ possible subarray value = global max - global min
        // একই subarray k বার choose করা যায়,
        // তাই answer = (gMax - gMin) * k
        return (long long)(gMax - gMin) * k;
    }
};
